### PR TITLE
chore(types): Add TSDoc comments to IInventoryService

### PR DESCRIPTION
**Pull request by shahednasser**
- Add TSDoc comments to the IInventoryService so that we can generate a reference later for the inventory module.
- Small fixes to the product module's TSDocs.

### CODE REVIEW

File: packages/types/src/inventory/service.ts [hunk 1]
```diff
@@ -17,163 +18,1124 @@ import { ModuleJoinerConfig } from "../modules-sdk"
...
+   *     relations: ["inventory_level"],
+   *     skip,
+   *     take
+   *   })
+   * 
+   *   // do something with the inventory items or return them
+   * }
+   * ```
+   * 
+   * You can also use the `$and` or `$or` properties of the `filter` parameter to use and/or conditions in your filters. For example:
```
Comment (pKorsholm) [inline packages/types/src/inventory/service.ts]: this way of querying is not supported in the inventory module due to typeorm

File: packages/types/src/inventory/service.ts [hunk 2]
```diff
@@ -17,163 +18,1124 @@ import { ModuleJoinerConfig } from "../modules-sdk"
...
+   *     relations: ["inventory_item"],
+   *     skip,
+   *     take
+   *   })
+   * 
+   *   // do something with the reservation items or return them
+   * }
+   * ```
+   * 
+   * You can also use the `$and` or `$or` properties of the `filter` parameter to use and/or conditions in your filters. For example:
```
Comment (pKorsholm) [inline packages/types/src/inventory/service.ts]: also here

File: packages/types/src/inventory/service.ts [hunk 3]
```diff
@@ -17,163 +18,1124 @@ import { ModuleJoinerConfig } from "../modules-sdk"
...
+   *     relations: ["inventory_item"],
+   *     skip,
+   *     take
+   *   })
+   * 
+   *   // do something with the inventory levels or return them
+   * }
+   * ```
+   * 
+   * You can also use the `$and` or `$or` properties of the `filter` parameter to use and/or conditions in your filters. For example:
```
Comment (pKorsholm) [inline packages/types/src/inventory/service.ts]: here

File: packages/types/src/inventory/service.ts [hunk 4]
```diff
@@ -17,163 +18,1124 @@ import { ModuleJoinerConfig } from "../modules-sdk"
...
   createReservationItem(
     input: CreateReservationItemInput,
     context?: SharedContext
   ): Promise<ReservationItemDTO>
 
-  // TODO make it bulk
+  /**
+   * @privateRemarks
+   * TODO make it bulk
+   *
```
Comment (pKorsholm) [inline packages/types/src/inventory/service.ts]:
```suggestion
```

File: packages/types/src/inventory/service.ts [hunk 5]
```diff
@@ -17,163 +18,1124 @@ import { ModuleJoinerConfig } from "../modules-sdk"
...
   deleteReservationItem(
     reservationItemId: string | string[],
     context?: SharedContext
   ): Promise<void>
 
-  // TODO make it bulk
+  /**
+   * @privateRemarks
+   * TODO make it bulk
+   *
```
Comment (pKorsholm) [inline packages/types/src/inventory/service.ts]:
```suggestion
```

File: packages/types/src/inventory/service.ts [hunk 6]
```diff
@@ -17,163 +18,1124 @@ import { ModuleJoinerConfig } from "../modules-sdk"
...
+   *   // do something with the reservation items or return them
+   * }
+   */
   createReservationItems(
     input: CreateReservationItemInput[],
     context?: SharedContext
   ): Promise<ReservationItemDTO[]>
 
+  /**
+   * This method is used to create inventory item.
```
Comment (pKorsholm) [inline packages/types/src/inventory/service.ts]:
```suggestion
   * This method is used to create an inventory item.
```

File: packages/types/src/inventory/common.ts [hunk 7]
```diff
@@ -267,16 +350,42 @@ export type CreateInventoryLevelInput = {
...
   incoming_quantity?: number
 }
 
+/**
+ * @interface
+ * 
+ * The attributes to update in an inventory level. The inventory level is identified by the IDs of its associated inventory item and location.
+ * 
+ * @prop inventory_item_id - The ID of the associated inventory level.
+ * @prop location_id - The ID of the associated location.
```
Comment (pKorsholm) [inline packages/types/src/inventory/common.ts:377]: question: should we include `stocked_quantity` and `incoming_quantity` in this?

File: packages/types/src/inventory/common.ts [hunk 8]
```diff
@@ -202,6 +202,20 @@ export type InventoryLevelDTO = {
...
+ * @prop line_item_id - Filter reservation items by the ID of their associated line item.
+ * @prop inventory_item_id - Filter reservation items by the ID of their associated inventory item.
+ * @prop location_id - Filter reservation items by the ID of their associated location.
+ * @prop description - Description filters to apply on the reservation items' `description` attribute.
+ * @prop created_by - The "created by" values to filter reservation items by.
+ * @prop quantity - Filters to apply on the reservation items' `quantity` attribute.
+ */
 export type FilterableReservationItemProps = {
   id?: string | string[]
   type?: string | string[]
```
Comment (pKorsholm) [inline packages/types/src/inventory/common.ts]: We don't have reservation types anymore
```suggestion
 * @prop id - The IDs to filter reservation items by.
 * @prop line_item_id - Filter reservation items by the ID of their associated line item.
 * @prop inventory_item_id - Filter reservation items by the ID of their associated inventory item.
 * @prop location_id - Filter reservation items by the ID of their associated location.
 * @prop description - Description filters to apply on the reservation items' `description` attribute.
 * @prop created_by - The "created by" values to filter reservation items by.
 * @prop quantity - Filters to apply on the reservation items' `quantity` attribute.
 */
export type FilterableReservationItemProps = {
  id?: string | string[]
```

**Review by pKorsholm — APPROVED** [review]
lgtm, just a couple of nits

File: packages/types/src/inventory/common.ts (hunk 7 above)
Comment (shahednasser) [reply packages/types/src/inventory/common.ts:377]: It includes the TSDoc comments for them defined in `UpdateInventoryLevelInput` by default.

**Pull request merged**